sdks/java/core/src/main/java/org/apache/beam/sdk/util/ReduceFnRunner.java

```diff
@@ -211,7 +213,9 @@ public ReduceFnRunner(
       PipelineOptions options) {
     this.key = key;
     this.timerInternals = timerInternals;
-    this.paneInfoTracker = new PaneInfoTracker(timerInternals);
+    this.paneInfoTracker = GlobalWindows.INSTANCE.isCompatible(windowingStrategy.getWindowFn()) ?
```

Comment:
A few ways I think to rejigger this:

1. `PaneInfoTrackers.forWindowFn(windowingStrategy.getWindowFn())` and put the logic in there.
    - and that logic should probably do an `instanceof`, since `isCompatible` has fuzzier meaning
2. `PaneInfoTrackers.forWindowingStrategy(windowingStrategy)` is a somewhat better abstraction, but less future-proof since `WindowingStrategy` is a potentially weird blob we ship and might dismantle eventually.